test/negativebinomial.jl

```diff
@@ -1,17 +1,37 @@
 using Distributions
+using ChainRulesTestUtils
+using FiniteDifferences
 using Test, ForwardDiff
 
-# Currently, most of the tests for NegativeBinomail are in the "ref" folder.
+# Currently, most of the tests for NegativeBinomial are in the "ref" folder.
 # Eventually, we might want to consolidate the tests here
 
-mydiffp(r, p, k) = r/p - k/(1 - p)
-
-@testset "NegativeBinomial r=$r, p=$p, k=$k" for
-    p in exp10.(-10:0) .- eps(), # avoid p==1 since it's not differentiable
-        r in exp10.(range(-10, stop=2, length=25)),
-            k in (0, 1, 2, 4, 8, 16, 32, 64, 128, 256, 512, 1024)
-
-    @test ForwardDiff.derivative(_p -> logpdf(NegativeBinomial(r, _p), k), p) ≈ mydiffp(r, p, k) rtol=1e-12 atol=1e-12
+@testset "NegativeBinomial" begin
+    @testset "ForwardDiff and rrule" begin
+        rs = -log.(rand(25))
+        ps = rand(25)
+        ks = (0, 1, 2, 4, 8, 16, 32, 64, 128, 256, 512, 1024)
+        @testset "r=$r" for r in rs
+            @testset "p=1, k=0" begin
```

Comment:
There's already a testset for the special case `p == 1` below, it might be better to move the tests there.

Reply:
I moved those tests up